appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -2168,40 +2173,89 @@ public boolean onCreateOptionsMenu(Menu menu) {
+      List<String> menuItems = new ArrayList();
+      addExitButtonToMenu(menu, menuItems);
+      addAboutInfoToMenu(menu, menuItems);
+      optionsMenuItems = YailList.makeList(menuItems);
+    } else {
+      MenuItems(optionsMenuItems);
+    }
     for (OnCreateOptionsMenuListener onCreateOptionsMenuListener : onCreateOptionsMenuListeners) {
       onCreateOptionsMenuListener.onCreateOptionsMenu(menu);
     }
     return true;
   }
 
-  public void addExitButtonToMenu(Menu menu) {
-    MenuItem stopApplicationItem = menu.add(Menu.NONE, Menu.NONE, Menu.FIRST,
-    "Stop this application")
+  public void addExitButtonToMenu(Menu menu, List<String> menuItems) {
+    String title = "Stop this application";
+    menuItems.add(title);
+    MenuItem stopApplicationItem = menu.add(Menu.NONE, Menu.NONE, 1, title)
     .setOnMenuItemClickListener(new OnMenuItemClickListener() {
       public boolean onMenuItemClick(MenuItem item) {
+        MenuItemSelected(item.getOrder(), item.getTitle().toString());
         showExitApplicationNotification();
         return true;
       }
     });
     stopApplicationItem.setIcon(android.R.drawable.ic_notification_clear_all);
   }
 
-  public void addAboutInfoToMenu(Menu menu) {
-    MenuItem aboutAppItem = menu.add(Menu.NONE, Menu.NONE, 2,
-    "About this application")
+  public void addAboutInfoToMenu(Menu menu, List<String> menuItems) {
+    String title = "About this application";
+    menuItems.add(title);
+    MenuItem aboutAppItem = menu.add(Menu.NONE, Menu.NONE, 2, title)
     .setOnMenuItemClickListener(new OnMenuItemClickListener() {
       public boolean onMenuItemClick(MenuItem item) {
+        MenuItemSelected(item.getOrder(), item.getTitle().toString());
         showAboutApplicationNotification();
         return true;
       }
     });
     aboutAppItem.setIcon(android.R.drawable.sym_def_app_icon);
   }
 
+  /**
+   * Set a list of text elements as options menu items
+   *
+   * @param itemsList a YailList containing the strings to be added to the menu
+   */
+  @SimpleProperty(description="List of text elements to show as items in the menu.  " +
+                "This will signal an error if the elements are not text strings.",
+      category = PropertyCategory.BEHAVIOR)
+  public void MenuItems(YailList itemsList) {
+    optionsMenuItems = ElementsUtil.elements(itemsList, "Menu");
+    if (optionsMenu == null) {
+      return;   // fill menu once onCreateOptionsMenu is executed
+    }
+    String[] items = optionsMenuItems.toStringArray();
+    optionsMenu.clear();
+    for (int i = 0; i < items.length; i++) {
+      optionsMenu.add(Menu.NONE, Menu.NONE, i+1, items[i])
+      .setOnMenuItemClickListener(new OnMenuItemClickListener() {
+        public boolean onMenuItemClick(MenuItem item) {
+          MenuItemSelected(item.getOrder(), item.getTitle().toString());
+          return true;
+        }
+      });
+    }
+  }
+
+  /**
+   * MenuItems property getter method
```

Comment:
This description will become the tooltip on the block, so you should make sure that it is descriptive as to what it might contain. See also my comment below about returning a copy of the list.